File: entity/Animal.cs
namespace AnimalRestAPI.entity;

public class Animal
{
    public int Id { get; set; }
    public string Name { get; set; }
    public Category Category { get; set; }
    public string Breed { get; set; }
    public string Color { get; set; }
    public List<Visit> Visits { get; set; }
}

File: Xd/src/main/java/pl/pomoku/xd/Animal.java
package pl.pomoku.xd;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Entity
@Data
@AllArgsConstructor
@NoArgsConstructor
public class Animal {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;
    private String name;
    @Enumerated(EnumType.ORDINAL)
    private AnimalCategory category;
    private String bread;
    @Enumerated(EnumType.ORDINAL)
    private Color color;
    @OneToMany(cascade = CascadeType.ALL, fetch = FetchType.EAGER)
    private List<Visit> visits;
}
